osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/JreEncryptionProvider.java

```diff
@@ -72,15 +77,28 @@ public String getFormat() {
 
             @Override
             public byte[] getEncoded() {
-                return key;
+                return JreEncryptionProvider.this.key;
             }
         };
     }
 
+    @Override
     protected AlgorithmParameterSpec getAlgorithmParameterSpec() {
         return new IvParameterSpec(IV);
     }
 
+    @Override
+    public byte[] generateAes128BitsSecret(String keyReference) throws EncrypterException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.opensmartgridplatform.shared.exceptionhandling.EncrypterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)